modules/ROOT/pages/fd-iterating.adoc

```diff
@@ -0,0 +1,112 @@
+
+For the Parallel For Each component, the value of a variable that enters the scope is the same at the start of the processing of each variable. New variables or modifications of already existing variables during the processing of one element are not visible during the processing of another element. Moreover, no changes to variables with the scope of a Parellel For-Each component are available to the subsequent cards in the flow.
+
+Suppose that, as in the previous example, you use a Set Variable component just before the scope of a Parallel For-Each component. Within the Set Variable component, you set the variable `NumOfElements` to 0. You then use a Set Variable component in the Parallel For-Each scope, using the same variable, that specifies a DataWeave script that increments the value by 1 every time it is called. For each parallel process that the component runs, the highest value reached by `NumOfElements` is 1. And if subsequent cards in the flow use `NumOfElements`, its initial value after the Parallel For-Each component finishes processing the collection is 0, which is the same value it had before the Parallel For-Each component started.
+
+
+== Error Handling
+
+Both the For-Each component and Parallel For-Each component stop processing a collection if an element causes any component, connector, or module in a scope to throw an exception. In addition, the Parallel For-Each component stops processing a collection if one of the parallel processes times out while waiting for an open connection to a system. You can set the maximum length of time that a parallel process can wait.
+
+When the processing of a collection stops because of an exception, Flow Designer displays a red circle on the card that threw the exception. When the processing of a collection stops because of a timeout, Flow Designer displays a red circle on the Parallel For-Each component's card. You can open the *Logs* pane to read the error message.
+
+== Configuring a For-Each Component
+
+When you configure this component, you must specify values in these fields:
+
+Collection:: An expression that returns a Java collection, object array, map, or DOM nodes. The default is `payload`.
+
+Batch Size:: Partitions the collection into sub-collections of the specified number of elements. For example, if a collection has 200 elements and you set the batch size to 50 elements, the component processes 4 batches of 50 elements each.
+
+Root Message Variable Name:: Name of the property that stores the entire message -- payload, variables, and attributes -- that was passed to a For-Each component from the previous card in a flow. 
```

Comment:
GLOBALLY replace dashes or em-dashes formed using spaces and hyphens with punctuation that has the same or similar meaning. In this case, I would use parentheses:
```suggestion
Root Message Variable Name:: Name of the property that stores the entire message (payload, variables, and attributes) that was passed to a For-Each component from the previous card in a flow. 
```